From 90bf47208c0db7615c00d0541169f5911bac2be0 Mon Sep 17 00:00:00 2001
From: George Kashperko <george@znau.edu.ua>
Date: Thu, 16 Mar 2023 15:12:33 +0530
Subject: [PATCH 037/281] OpenWrt:
 461-mtd-cfi_cmdset_0002-add-buffer-write-cmd-timeout.patch

Issue map read after Write Buffer Load command to ensure chip is ready to receive data.

Signed-off-by: George Kashperko <george@znau.edu.ua>

Change-Id: Idd9670bc64e5cfb6c6f021f4388520524912488b
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
---
 drivers/mtd/chips/cfi_cmdset_0002.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/mtd/chips/cfi_cmdset_0002.c b/drivers/mtd/chips/cfi_cmdset_0002.c
index a6692d72f24b..13694138a8b8 100644
--- a/drivers/mtd/chips/cfi_cmdset_0002.c
+++ b/drivers/mtd/chips/cfi_cmdset_0002.c
@@ -2052,6 +2052,7 @@ static int __xipram do_write_buffer(struct map_info *map, struct flchip *chip,
 
 	/* Write Buffer Load */
 	map_write(map, CMD(0x25), cmd_adr);
+	(void) map_read(map, cmd_adr);
 
 	chip->state = FL_WRITING_TO_BUFFER;
 
-- 
2.17.1

